### DISCUSSION TITLE
How to list source files not relatated to sequence points.

### DISCUSSION THREAD

**Discussion by ths11** [jbevain/cecil#831] [Q&A]
I would like to extract the list of all source files indexed in a PDB.
By invoking `PdbReader.GetSourceFiles()` I only get the list of files referenced by some sequence point, but source files related to simple interfaces or enumerations (example below) are not listed even if the PDB contains a stream named "/SRC/FILES/<src_file>".

Looking at the code (especially `PdbFile.PdLoadFunctions()`) I confirmed that the information is not loaded by default.

Is there any API that allows access to the full source list (as the one displayed by the MS `srctool.exe` PDB utility ) or is coding work needed? 
In the later case, would a merge request be appreciated?

Example of source file ignored by `PdbReader.GetSourceFiles()`:
`
namespace Foo
{
    public enum Bar {
        A_CNT_0 = 0,
        A_CNT_1 = 1
    };
}
`

**Comment by jbevain** [comment]
Unfortunately there's nothing in Cecil (or in the managed Pdb reader we're using) to achieve this easily. You'd need to modify the managed pdb reader I'm afraid.
